library/Businessprocess/BpConfig.php

```diff
@@ -290,9 +292,23 @@ public function setBackend(MonitoringBackend $backend)
     public function getBackend()
     {
         if ($this->backend === null) {
-            $this->backend = MonitoringBackend::instance(
-                $this->getBackendName()
-            );
+            if ($this->getBackendName() === '_icingadb' ||
+                (Module::exists('icingadb') && IcingadbSupport::useIcingaDbAsBackend())
+            ) {
+                if (! Module::exists('icingadb')) {
+                    throw new Exception('Icingadb module is not enabled.');
+                }
+
+                $this->backend = IcingaDbObject::fetchDb();
+            } else {
+                if (! Module::exists('monitoring') && Module::exists('icingadb')) {
+                    $this->backend = IcingaDbObject::fetchDb();
```

Comment:
This is redundant. `IcingadbSupport::useIcingaDbAsBackend()` is already true if monitoring is disabled.